atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -146,21 +167,47 @@ def run(self):
                 self.log.debug("release set explicitly so not incrementing")
                 if not is_scratch_build():
                     self.check_build_existence_for_explicit_release(component, version, release)
-                return
-
-        if self.append:
-            next_release = self.get_next_release_append(component, version, release)
-        elif is_scratch_build():
-            metadata = get_build_json().get("metadata", {})
-            next_release = metadata.get("name", "1")
-        else:
-            next_release = self.get_next_release_standard(component, version)
+                else:
+                    return
 
         # Always set preferred release label - other will be set if old-style
         # label is present
         release_label = labels.LABEL_NAMES[Labels.LABEL_TYPE_RELEASE][0]
 
-        # No release labels are set so set them
-        self.log.info("setting %s=%s", release_label, next_release)
-        # Write the label back to the file (this is a property setter)
-        dockerfile_labels[release_label] = next_release
+        if not release or self.append:
+            self.get_next_release_general(component, version, release, release_label,
+                                          dockerfile_labels)
+
+        if not self.predeclare_build or is_scratch_build():
+            return
+
+        for counter in range(KOJI_PREDECLARE_MAX_RETRIES + 1):
```

Comment:
IMO this whole pre-declare block could be extracted into a function for better readability

Reply:
moved to function